### DISCUSSION TITLE
Minimal body to start writing a new plugin

### DISCUSSION THREAD

**Discussion by piedagile** [pierreguillot/Camomile#246] [General]
I'm new to this thread, and I've just created my first minimal plugin to make sure I understood. I learned by working with the examples, it works with my DAW, but it took me a little time to prune out what is specific to each example of what is the minimum to embed. So I give here a minimum checklist and I submit it to your criticism.

For example, with a new effect called Gain:

PD patches: param.change.pd, param.get.pd, param.set.pd, program.get.pd : the same for any plugin

Infos.txt: free text

Gain.png: the Gain module background displayed by the DAW

Gain.txt: 
(1 minimum) param line ;  iolayout 2 2 | 2 0 (for Logic sidechain);  code for subtype (4 char); type effect; image Gain.png; description Infod.txt; manufacturer ME; compatibility 1.0.7;

Gain.pd: this Pure Dat Vanilla patch contains, at the minimum
- user graphic interface, here a simple slider connected by its send/receive symbols $0-gain-s $0-gain-r, and delimited by a Graph-On-Parent area
- connectors adc~ and dac~
- param.set 1 $0-gain-s
- param get 1 that apply gain to the output
- program.get 1 $0-gain-r
- param.change (I don't know its role, but must be there)

Then the structure is ready to apply the How to generate plugins procedure

Do you agree or I missed something? Thank you for advices.
Charles

**Comment by pierreguillot** [comment]
Thank you, Charles! I'm sure it will be helpful for new users! If you could share a zip file with all the content, it would be great!

**Reply by piedagile** [reply]
Yes, Pierre, I should have done it already !
[Gain.zip](https://github.com/pierreguillot/Camomile/files/6199707/Gain.zip)
